File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.java
import java.util.*;
class Solution {
    public int solution(int[] rank, boolean[] attendance) {
        int answer = 0;
        
        List<Integer> list = new ArrayList<>();

        for(int i = 0; i < rank.length; i++) {
            if(attendance[i]) {
                list.add(rank[i]);
            }else {
                rank[i] = 101;
                list.add(101);
            }
        }
        
        Arrays.sort(rank);

        answer = 10000 * list.indexOf(rank[0]) + 100 * list.indexOf(rank[1]) + list.indexOf(rank[2]);
        
        return answer;
    }
}

File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.py
def solution(rank, attendance):
    answer = 0
    
    rank = [[i,rank[i]] for i in range(len(rank)) if attendance[i]]
    
    # 익명함수
    rank = sorted(rank, key=lambda x: x[1])
    
    answer = 10000 * rank[0][0] + 100 * rank[1][0] + rank[2][0]
    
    return answer